test/parallel_api/ranges/reduce_by_key_sycl.pass.cpp

```diff
@@ -0,0 +1,109 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#include <oneapi/dpl/execution>
+#include <oneapi/dpl/algorithm>
+#include <oneapi/dpl/iterator>
+
+#include "support/test_config.h"
+#include "support/utils.h"
+
+#include <iostream>
+#include <vector>
+#include <cassert>
+
+#if TEST_DPCPP_BACKEND_PRESENT
+#include <CL/sycl.hpp>
+#endif // TEST_DPCPP_BACKEND_PRESENT
+
+#if TEST_DPCPP_BACKEND_PRESENT
+// Uncomment for extended logging
+// #define _ONEDPL_DEBUG_SYCL 1
+#endif // TEST_DPCPP_BACKEND_PRESENT
+
+int
+main()
+{
+#if TEST_DPCPP_BACKEND_PRESENT
+    auto policy = oneapi::dpl::execution::dpcpp_default;
+
+    sycl::usm_allocator<int, sycl::usm::alloc::shared> alloc(policy.queue());
+
+    // this line is fine -> [1], [6]
+    std::vector<int, decltype(alloc)> keys  ({ 1, 1, 1 }, alloc);
+    std::vector<int, decltype(alloc)> values({ 1, 2, 3 }, alloc);
+
+    // this won't produce output -> expected [0], [6], but evaluated [0], [1]
+    //std::vector<int, decltype(alloc)> keys  ({ 0, 0, 0 }, alloc);
```

Comment:
Is this a part of comment or forgotten commented "debug" code?

Reply:
Fixed